test_regress/t/TestSimulator.h

```diff
@@ -29,12 +29,14 @@ class TestSimulator {
         vpi_get_vlog_info(&m_info);
         if (0 == strcmp(m_info.product, "Verilator")) {
             m_simulators.verilator = true;
-        } else if (0 == strcmp(m_info.product, "Verilator")) {
+        } else if (0 == strcmp(m_info.product, "Icarus Verilog")) {
```

Comment:
Some stuff I noticed along the way wrt other simulators.  However, this isn't particularly germane to this patch so I can revert this file.

Reply:
This is fine, but ideally would push it separately, feel free to do that if you want and commit it once tests pass.  Similar for below.